### ISSUE TITLE
Potential recurrence of #13309

### ISSUE THREAD

**Issue by josalem** [dotnet/runtime#31900]
I'm experiencing this issue with .NET Core 3.1

This causes the Linux container to hit OOM. Running the app in VS in debugger has the same issue. 

Unfortunately, I cannot share my memory dump. But the summary is that I have the Azure Service Bus  lib in my app. The service bus lib tries to connect to the topic you specify. If the topic does not exists it keeps throwing exceptions indefinitely. This causes the AI to keep gobbling memory.

This also causes the container to fail to swap at release time bc it runs OOM.

_Originally posted by @nurevaSaeed in https://github.com/dotnet/runtime/issues/13309#issuecomment-583118907_

**Comment by josalem** [comment]
@nurevaSaeed, could you describe the scenario a little more?  Specifically
* What applications are running (including your own) inside your container or when you are debugging with VS?
* Are you tracing your application using `dotnet-trace` or using another .NET Diagnostics Tool when this happens?
* Are you using `DiagnosticListener` in your application?
* Are you using `prometheus-net.DotNetRuntime` in your app?
* What specifically makes you think you're seeing a recurrence of #13309, e.g., are you seeing a leak of buffers of the same size as described there?

Lastly, could you explain this a bit more:

> This causes the AI to keep gobbling memory.

Are you talking about Application Insights?

If possible, a minimal repro of the behavior, some output from the `!heap -stat` command showing the leak, or some other concrete evidence of the issue would be very helpful in triaging this.

**Comment by nurevaSaeed** [comment]
Sorry @josalem, I'd deleted my post before seeing your message. I will send an update with what you're asking soon. For now, I don't think my issue applies to this bug anymore.

**Comment by tommcdon** [comment]
@semyon2105 are you still experiencing the CPU issues mentioned in djluck/prometheus-net.DotNetRuntime#6?

**Comment by semyon2105** [comment]
@tommcdon yes, still seeing the same CPU increase in repro using the latest .NET SDK Docker image

**Comment by tommcdon** [comment]
@sywhang can you take a look?

**Comment by sywhang** [comment]
@semyon2105 What version of .NET Core 3.1 are you using?

**Comment by semyon2105** [comment]
@sywhang I'm using version 3.1.6

**Comment by sywhang** [comment]
Thanks for confirming that. Just to make sure, is https://github.com/semyon2105/leak-repro still a valid repro for your issue?

**Comment by semyon2105** [comment]
Yes, and I've just updated it to use the latest .NET image

**Comment by sywhang** [comment]
@semyon2105 Thanks. I tried running your repro and have some findings I'd like to share, but since the issue is unrelated to any runtime issues (including the one posted here), I will post my findings in the issue you initially reported it in the prometheus-net repo. 

@josalem I believe the recurrence that you saw here is related to the mmap fix I made back in 3.1.5 - I'm going to close this issue since we've addressed that issue.
